src/Compilers/Core/Portable/Compilation/IExpression.cs

```diff
@@ -828,6 +829,74 @@ public static BinaryOperandsKind GetBinaryOperandsKind(BinaryOperationKind kind)
         }
     }
 
+    public interface IConversion
+    {
+        /// <summary>
+        /// Returns true if the conversion exists, either as an implicit or explicit conversion.
+        /// </summary>
+        bool Exists { get; }
+
+        /// <summary>
+        /// Returns true if the conversion is an identity conversion.
+        /// </summary>
+        bool IsIdentity { get; }
+
+        /// <summary>
+        /// Returns true if the conversion is an implicit ('widening' in VB) conversion.
+        /// </summary>
+        bool IsImplicit { get; }
+
+        /// <summary>
+        /// True if and only if the conversion is an explicit ('narrowing' in VB) conversion.
+        /// </summary>
+        bool IsExplicit { get; }
+
+        /// <summary>
+        /// Returns true if the conversion is an implicit numeric conversion or explicit numeric conversion. 
+        /// </summary>
+        bool IsNumeric { get; }
+
+        /// <summary>
+        /// Returns true if the conversion is an implicit nullable conversion or explicit nullable conversion.
+        /// </summary>
+        bool IsNullable { get; }
+
+        /// <summary>
+        /// Returns true if the conversion is an implicit reference conversion or explicit reference conversion.
+        /// </summary>
+        bool IsReference { get; }
+
+        /// <summary>
+        /// Returns true if the conversion is an implicit user-defined conversion or explicit user-defined conversion.
+        /// </summary>
+        bool IsUserDefined { get; }
+
+        /// <summary>
+        /// Returns true if the conversion is an implicit default ('null' in C#, 'nothing' in VB) literal conversion.
+        /// </summary>
+        bool IsDefault { get; }
+
+        /// <summary>
+        /// Returns the method that defines the user defined conversion, if any..
+        /// </summary>
+        IMethodSymbol MethodSymbol { get; }
+    }
+
+    public static class ConversionExtensions
+    {
+        public static IConversion ClassifyConversion(
```

Comment:
Note: i wanted to originally add these to Compilation and SemanticModel.  However, this caused a source level break because we already have extensions with these names defined in the C# and VB compiler dlls.